src/core/ext/transport/chaotic_good/server_transport.cc

```diff
@@ -0,0 +1,334 @@
+    FrameHeader frame_header, BufferPair buffers) {
+  ClientFragmentFrame fragment_frame;
+  ScopedArenaPtr arena(acceptor_->CreateArena());
+  absl::Status status = transport_.DeserializeFrame(
+      frame_header, std::move(buffers), arena.get(), fragment_frame);
+  absl::optional<CallInitiator> call_initiator;
+  if (status.ok()) {
+    auto create_call_result =
+        acceptor_->CreateCall(*fragment_frame.headers, arena.release());
+    if (create_call_result.ok()) {
+      call_initiator.emplace(std::move(*create_call_result));
+      call_initiator->SpawnGuarded(
+          "server-write", [this, stream_id = frame_header.stream_id,
+                           call_initiator = *call_initiator]() {
+            return CallOutboundLoop(stream_id, call_initiator);
+          });
+    } else {
+      status = create_call_result.status();
+    }
+  }
+  return MaybePushFragmentIntoCall(std::move(call_initiator), std::move(status),
+                                   std::move(fragment_frame));
+}
+
+auto ChaoticGoodServerTransport::DeserializeAndPushFragmentToExistingCall(
+    FrameHeader frame_header, BufferPair buffers) {
+  absl::optional<CallInitiator> call_initiator =
+      LookupStream(frame_header.stream_id);
+  Arena* arena = nullptr;
+  if (call_initiator.has_value()) arena = call_initiator->arena();
+  ClientFragmentFrame fragment_frame;
+  absl::Status status = transport_.DeserializeFrame(
+      frame_header, std::move(buffers), arena, fragment_frame);
+  return MaybePushFragmentIntoCall(std::move(call_initiator), std::move(status),
+                                   std::move(fragment_frame));
+}
+
+auto ChaoticGoodServerTransport::TransportReadLoop() {
+  return Loop([this] {
+    return TrySeq(
+        transport_.ReadFrameBytes(),
+        [this](std::tuple<FrameHeader, BufferPair> frame_bytes) {
+          const auto& frame_header = std::get<0>(frame_bytes);
+          auto& buffers = std::get<1>(frame_bytes);
+          return Switch(
+              frame_header.type,
+              Case(FrameType::kSettings,
+                   []() -> absl::Status {
+                     return absl::InternalError("Unexpected settings frame");
+                   }),
+              Case(FrameType::kFragment,
+                   [this, &frame_header, &buffers]() {
+                     return If(
+                         frame_header.flags.is_set(0),
+                         [this, &frame_header, &buffers]() {
+                           return DeserializeAndPushFragmentToNewCall(
+                               frame_header, std::move(buffers));
+                         },
+                         [this, &frame_header, &buffers]() {
+                           return DeserializeAndPushFragmentToExistingCall(
+                               frame_header, std::move(buffers));
+                         });
+                   }),
+              Case(FrameType::kCancel,
```

Comment:
When does the Cancel frame actually get sent ? I don't see a corresponding send in the ClientTransport. Is this something we plan to add later ?